backend/src/v5/processors/teamspaces/audits.js

```diff
@@ -0,0 +1,68 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { createResponseCode, templates } = require('../../utils/responseCodes');
+const { PassThrough } = require('stream');
+const archiver = require('archiver');
+archiver.registerFormat('zip-encrypted', require('archiver-zip-encrypted'));
+const { templates: emailTemplates } = require('../../services/mailer/mailer.constants');
+const { generateHashString } = require('../../utils/helper/strings');
+const { getActionLog } = require('../../models/teamspaces.audits');
+const { getUserByUsername } = require('../../models/users');
+const { sendEmail } = require('../../services/mailer');
+
+const Audit = {};
+
+const createAuditLogArchive = async (actions) => {
+	try {
+		const password = generateHashString();
+		const stream = new PassThrough();
+
+		const archive = archiver.create('zip-encrypted', {
+			zlib: { level: 1 },
+			encryptionMethod: 'aes256',
+			password,
+		});
+
+		const content = `[${actions.map((action) => JSON.stringify(action)).join(',')}]`;
```

Comment:
As it stands, I think we'll be getting the dates in string format. we want it in epoch timestamps.

Does project ID look right here? they should be stored as UUID so i'd image we're going to get base64 jibberish?

I think we should run it through a yup cast (so we're also future proof if anything else needs a conversion)

Reply:
This, once again is not great for memory if actions is large. you're still storing the content in a big string.

Utiilise the streaming functionality and write them into the stream one by one.